detekt-rules/src/test/kotlin/org/example/detekt/PublicDataClassConstructorWithValueParametersTest.kt

```diff
@@ -0,0 +1,106 @@
+package org.example.detekt
+
+import io.gitlab.arturbosch.detekt.api.Config
+import io.gitlab.arturbosch.detekt.rules.KotlinCoreEnvironmentTest
+import io.gitlab.arturbosch.detekt.test.compileAndLintWithContext
+import io.kotest.matchers.collections.shouldHaveSize
+import org.jetbrains.kotlin.cli.jvm.compiler.KotlinCoreEnvironment
+import org.junit.jupiter.api.Test
+
+@KotlinCoreEnvironmentTest
+internal class PublicDataClassConstructorWithValueParametersTest(
+  private val env: KotlinCoreEnvironment
+) {
+  @Test
+  fun `reports data classes that have value class parameters without companion static constructor`() {
+    val code =
+      """
+        import kotlin.jvm.JvmInline
+        @JvmInline value class  Bar(val value: String)
+        data class Foo(val bar: Bar)
```

Comment:
Is the problem here the same as Rule 1?, that we are using an `inline value class` as a property in a data class?
Do we need need to test that data classes have static methods? I think instead of testing that if there is a static factory we should just disallow value classes altogether and forbid their usage.